File: domain/src/main/scala/com/anjunar/technologyspeaks/codemirror/CodeMirrorHTML.scala
package com.anjunar.technologyspeaks.codemirror

import jakarta.persistence.{Basic, Column, Entity, Lob}

import scala.compiletime.uninitialized

@Entity
class CodeMirrorHTML extends AbstractCodeMirrorFile {

  @Basic
  @Lob
  @Column(columnDefinition = "text")
  var content: String = uninitialized

}

File: application/src/main/typescript/packages/library/src/components/inputs/codemirror/domain/CodeMirrorHTML.ts
import {AbstractCodeMirrorFile} from "./AbstractCodeMirrorFile";
import Basic from "../../../../mapper/annotations/Basic";
import {Entity} from "../../../../mapper";
import {CodeMirrorContent} from "./CodeMirrorContent";

@Entity("CodeMirrorHTML")
export class CodeMirrorHTML extends AbstractCodeMirrorFile implements CodeMirrorContent{

    $type = "CodeMirrorHTML"

    @Basic()
    content : string

}